provider/pkg/gen/utilities.go

```diff
@@ -32,8 +47,13 @@ func ToLowerCamel(s string) string {
 	if uppercaseAcronym[s] {
 		s = strings.ToLower(s)
 	}
-	if r := rune(s[0]); r == '_' {
-		s = s[1:]
+	r := []rune(s)
+	if prefix, found := commonPrefixReplacements[r[0]]; found {
+		if len(r) > 1 {
+			s = prefix + strings.ToUpper(string(r[1])) + string(r[2:])
```

Comment:
```suggestion
			s = prefix + strings.ToUpper(string(r[1])) + r[2:]
```
I think this may work

Reply:
Nope .. `invalid operation: mismatched types string and []rune` - think it needs the explicit conversion

Reply:
How does this work then? https://github.com/pulumi/pulumi-azure-native/blob/95c2dc8b77fc3675ed8328374a1e18ee2af7608f/provider/pkg/gen/utilities.go#L59

Reply:
Ah, I missed `r := []rune(s)`! I see.